CoseHandler/CoseHandler.cs

```diff
@@ -55,11 +58,12 @@ public static ReadOnlyMemory<byte> Sign(
         X509Certificate2 certificate,
         bool embedSign = false,
         FileInfo? signatureFile = null,
-        string contentType = CoseSign1MessageFactory.DEFAULT_CONTENT_TYPE)
+        string contentType = CoseSign1MessageFactory.DEFAULT_CONTENT_TYPE,
+        ICoseHeaderExtender? headerExtender = null)

```

Comment:
Why do some Sign overloads get the headerExtender option and not others?
It looks like you have it on this overload and on the ones that take a signing key provider.

Reply:
I had the same question as I did not find many overloads not being used. Any reason why they were created?

Reply:
They were created for user convenience because CoseHandler is intended to be the main entry point for common API scenarios. As such I wanted to give it flexibility and ease of use, so the user could supply their payload, signatures, and certificates in any combination of supported formats and have it work.

Reply:
Added it to all the overloads that were missing it.